stdlib/public/core/CMakeLists.txt

```diff
@@ -246,34 +246,42 @@ else()
   # effort has been completed.
   #set(LINK_FLAGS
   #  -Wl,--whole-archive swiftRuntime -Wl,--no-whole-archive)
-  if("${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
-    list(APPEND swift_core_private_link_libraries
-      ${SWIFT_${SWIFT_PRIMARY_VARIANT_SDK}_${SWIFT_PRIMARY_VARIANT_ARCH}_ICU_UC}
-      ${SWIFT_${SWIFT_PRIMARY_VARIANT_SDK}_${SWIFT_PRIMARY_VARIANT_ARCH}_ICU_I18N})
-  else()
+  if(NOT "${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
     list(APPEND swift_core_private_link_libraries -licui18nswift -licuucswift -licudataswift)
   endif()
 endif()
 
-if(SWIFT_PRIMARY_VARIANT_SDK STREQUAL ANDROID)
-  # workaround for libatomic needing to be after object files for swiftCore.so
-  list(APPEND swift_core_private_link_libraries atomic)
-elseif(SWIFT_PRIMARY_VARIANT_SDK STREQUAL CYGWIN)
+function(swift_core_private_libraries sdk arch libraries_var_name)
+  set(private_link_libraries)
+  if(NOT sdk IN_LIST SWIFT_APPLE_PLATFORMS AND "${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
+    list(APPEND private_link_libraries ${SWIFT_${sdk}_${arch}_ICU_UC}
+      ${SWIFT_${sdk}_${arch}_ICU_I18N})
+  endif()
+
+  if(sdk STREQUAL ANDROID)
+    # workaround for libatomic needing to be after object files for swiftCore.so
+    list(APPEND private_link_libraries atomic)
+  elseif(sdk STREQUAL FREEBSD)
+    find_library(EXECINFO_LIBRARY execinfo)
```

Comment:
I don't know if `find_library` will work when cross-compiling, so this will need to be pulled out like `CYGWIN` below if not.

Reply:
`find_library` is completely capable of working with cross-compilation, please read up on CMake's cross-compilation support, its pretty well documented.

Reply:
I know, I subsequently used it [when cross-compiling the Swift stdlib for Android](https://github.com/termux/termux-packages/pull/5169/files#diff-bd95e8ae22237283c1199931c04d509aR12). Anyway, nobody is likely to ever cross-compile the Swift stdlib to BSD, so this point is moot.